org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/editor/SemanticHighlightingTest.java

```diff
@@ -11,49 +11,49 @@
 import java.util.Set;
 
 import org.eclipse.jface.text.TypedRegion;
-import org.eclipse.xtext.XtextRuntimeModule;
-import org.eclipse.xtext.XtextStandaloneSetup;
 import org.eclipse.xtext.ide.editor.syntaxcoloring.IHighlightedPositionAcceptor;
 import org.eclipse.xtext.ide.editor.syntaxcoloring.ISemanticHighlightingCalculator;
 import org.eclipse.xtext.junit4.AbstractXtextTests;
 import org.eclipse.xtext.resource.XtextResource;
-import org.eclipse.xtext.ui.shared.SharedStateModule;
+import org.eclipse.xtext.testing.InjectWith;
+import org.eclipse.xtext.testing.XtextRunner;
 import org.eclipse.xtext.util.CancelIndicator;
-import org.eclipse.xtext.util.Modules2;
-import org.eclipse.xtext.xtext.ui.Activator;
+import org.eclipse.xtext.xtext.ui.XtextUiInjectorProvider;
 import org.eclipse.xtext.xtext.ui.editor.syntaxcoloring.SemanticHighlightingConfiguration;
-import org.eclipse.xtext.xtext.ui.internal.XtextUIModuleInternal;
 import org.junit.Test;
+import org.junit.runner.RunWith;
 
 import com.google.common.collect.Sets;
-import com.google.inject.Guice;
+import com.google.inject.Inject;
 import com.google.inject.Injector;
 
 /**
  * @author Sebastian Zarnekow - Initial contribution and API
  */
+@RunWith(XtextRunner.class)
+@InjectWith(XtextUiInjectorProvider.class)
 public class SemanticHighlightingTest extends AbstractXtextTests implements IHighlightedPositionAcceptor {
-	
+
+	@Inject
+	private Injector injector;
+
+	@Inject
+	private ISemanticHighlightingCalculator calculator;
+
 	private Set<TypedRegion> expectedRegions;
-	
+
 	@Override
 	public void setUp() throws Exception {
 		super.setUp();
-		with(new XtextStandaloneSetup() {
-			@Override
-			public Injector createInjector() {
-				return Guice.createInjector(Modules2.mixin(new XtextRuntimeModule(),new XtextUIModuleInternal(Activator.getDefault()), new SharedStateModule()));
-			}
-		});
 		expectedRegions = Sets.newLinkedHashSet();
+		setInjector(injector);
```

Comment:
You can override `setInjector()` and mark it with `@Inject` which would allow to remove the injector field above.

Reply:
Fixed.